pinot-core/src/main/java/org/apache/pinot/core/data/table/ConcurrentIndexedTable.java

```diff
@@ -101,31 +108,51 @@ public int size() {
 
   @Override
   public Iterator<Record> iterator() {
+    if (_sort && CollectionUtils.isNotEmpty(_orderBy)) {
+      List<Record> sortedList = new ArrayList<>(_lookupMap.values());
+      sortedList.sort(_orderByComparator);
+      return sortedList.iterator();
+    }
     return _lookupMap.values().iterator();
   }
 
   private void resize(int trimToSize) {
 
     if (_lookupMap.size() > trimToSize) {
```

Comment:
collect some metrics in this table such as num resizes, time taken to resize etc that can be printed/emitted in finish method to understand the perf bottlenecks

Reply:
Done. Printed num resizes, total time resizing, and avg resize time in finish.